unit_tests/testParameterLists/testParameterLists.f90

```diff
@@ -4653,6 +4677,2790 @@ SUBROUTINE testSDKa4()
 ENDSUBROUTINE testSDKa4
 !
 !-------------------------------------------------------------------------------
+!Test 5-D Array SNK support
+SUBROUTINE testSNKa5()
+  LOGICAL(SBK) :: bool
+  CHARACTER(LEN=EXCEPTION_MAX_MESG_LENGTH) :: msg,refmsg
+  ALLOCATE(valsnka5(2,2,2,2,1))
+  valsnka5(1,1,1,:,1)=1_SNK
+  valsnka5(2,1,1,:,1)=2_SNK
+  valsnka5(1,2,1,:,1)=3_SNK
+  valsnka5(2,2,1,:,1)=4_SNK
+  valsnka5(1,1,2,:,1)=5_SNK
+  valsnka5(2,1,2,:,1)=6_SNK
+  valsnka5(1,2,2,:,1)=7_SNK
+  valsnka5(2,2,2,:,1)=8_SNK
+  !test init
+
+  COMPONENT_TEST('%init(...)')
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat 1')
+  ASSERT(LEN(testParam%name) == 0,'%name 1')
+  ASSERT(LEN(testParam%datatype) == 0,'%datatype 1')
+  ASSERT(LEN(testParam%description) == 0,'%description 1')
+  CALL testParam%init('testError->testSNKa5',valsnka5,'The numbers 5, 7, 6, & 8')
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat 2')
+  ASSERT(LEN(testParam%name) == 0,'%name 2')
+  ASSERT(LEN(testParam%datatype) == 0,'%datatype 2')
+  ASSERT(LEN(testParam%description) == 0,'%description 2')
+  msg=eParams%getLastMessage()
+  refmsg='#### EXCEPTION_ERROR #### - PARAMETERLISTS::init_ParamType_SNK_a5'// &
+      ' - "->" symbol is not allowed in name!'
+  ASSERT(TRIM(msg) == TRIM(refmsg),'init bad symbol error')
+  CALL testParam%init('testSNKa5',valsnka5,'The numbers 5, 7, 6, & 8')
+  ASSERT(LEN(testParam%name) == 0,'%name 4')
+  ASSERT(LEN(testParam%datatype) == 0,'%datatype 4')
+  ASSERT(LEN(testParam%description) == 0,'%description 4')
+  ASSERT(ASSOCIATED(testParam%pdat),'%pdat 4')
+  ASSERT(testParam%pdat%name == 'testSNKa5','%pdat%name')
+  ASSERT(testParam%pdat%datatype == '5-D ARRAY INTEGER(SNK)','%pdat%datatype')
+  ASSERT(testParam%pdat%description == 'The numbers 5, 7, 6, & 8','%pdat%description')
+  CALL testParam%init('testError',valsnka5)
+  msg=eParams%getLastMessage()
+  refmsg='#### EXCEPTION_ERROR #### - PARAMETERLISTS::init_ParamType_SNK_a5'// &
+      ' - parameter is already initialized! Use set method!'
+  ASSERT(TRIM(msg) == TRIM(refmsg),'double init error')
+
+  !Test clear
+  COMPONENT_TEST('%clear()')
+  CALL testParam%clear()
+  ASSERT(LEN(testParam%name) == 0,'%name')
+  ASSERT(LEN(testParam%datatype) == 0,'%datatype')
+  ASSERT(LEN(testParam%description) == 0,'%description')
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat')
+
+  COMPONENT_TEST('%edit(...)')
+  CALL testParam%init('testSNKa5',valsnka5,'The numbers 5, 7, 6, & 8')
+  CALL testParam%edit(OUTPUT_UNIT,0)
+  CALL testParam%clear()
+  CALL testParam%init('testSNKa5',valsnka5)
+  CALL testParam%edit(OUTPUT_UNIT,0)
+  CALL testParam%clear()
+  CALL testParam%edit(OUTPUT_UNIT,0)
```

Comment:
What is this component test even doing?

Reply:
I guess just making sure it doesn't bomb hard while writing.  You're silly if you think I actually read any of this code though.